src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -389,18 +428,27 @@ bool CoffNativeCodeManager::UnwindStackFrame(MethodInfo *    pMethodInfo,
     memset(&contextPointers, 0xDD, sizeof(contextPointers));
 #endif
 
-#define FOR_EACH_NONVOLATILE_REGISTER(F) \
-    F(Rax) F(Rcx) F(Rdx) F(Rbx) F(Rbp) F(Rsi) F(Rdi) F(R8) F(R9) F(R10) F(R11) F(R12) F(R13) F(R14) F(R15)
+#ifdef _TARGET_X86_
+    #define FOR_EACH_NONVOLATILE_REGISTER(F) \
+        F(E, ax) F(E, cx) F(E, dx) F(E, bx) F(E, bp) F(E, si) F(E, di)
+    #define WORDPTR PDWORD
+#else
+    #define FOR_EACH_NONVOLATILE_REGISTER(F) \
+        F(R, ax) F(R, cx) F(R, dx) F(R, bx) F(R, bp) F(R, si) F(R, di) \
+        F(R, 8) F(R, 9) F(R, 10) F(R, 11) F(R, 12) F(R, 13) F(R, 14) F(R, 15)
+    #define WORDPTR PDWORD64
+#endif
 
-#define REGDISPLAY_TO_CONTEXT(reg) \
-    contextPointers.reg = (PDWORD64) pRegisterSet->p##reg; \
-    if (pRegisterSet->p##reg != NULL) context.reg = *(pRegisterSet->p##reg);
+#define REGDISPLAY_TO_CONTEXT(prefix, reg) \
+    contextPointers.prefix####reg = (WORDPTR) pRegisterSet->pR##reg; \
+    if (pRegisterSet->pR##reg != NULL) context.prefix##reg = *(pRegisterSet->pR##reg);
 
-#define CONTEXT_TO_REGDISPLAY(reg) \
-    pRegisterSet->p##reg = (PTR_UIntNative) contextPointers.reg;
+#define CONTEXT_TO_REGDISPLAY(prefix, reg) \
+    pRegisterSet->pR##reg = (PTR_UIntNative) contextPointers.prefix####reg;
 
     FOR_EACH_NONVOLATILE_REGISTER(REGDISPLAY_TO_CONTEXT);
 
+#ifndef _TARGET_X86_
```

Comment:
This can use `PORTABILITY_ASSERT` under the x86 ifdef.

Reply:
Landed in 63a654a.